lib/net/xhr.js

```diff
@@ -45,9 +45,16 @@ function send(url, options) {
         onHeaders = options.onHeaders || noop,
         errorFn = options.error || noop,
         completeFn = options.complete || noop,
+        disableContentType = options.disableContentType || false,
+        onUploadProgress = options.onUploadProgress,
         addContentHeader = method === 'POST',
         timer;
 
+    // Not all browsers support upload events
+    if (onUploadProgress != null && xhr.upload) {
```

Comment:
Can we do a `typeof` check to see if it is a function, instead of just checking `null`